dts/x86/arduino_101.dts

```diff
@@ -0,0 +1,29 @@
+/dts-v1/;
+
+#include "intel_curie.dtsi"
+
+/ {
+	model = "Intel Curie Arduino 101 board";
+	compatible = "intel,quark";
+
+	aliases {
+		uart_0 = &uart0;
+		uart_1 = &uart1;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart1;
+	};
+};
+
+&uart0 {
+	status = "ok";
+	current-speed = <115200>;
+};
+
```

Comment:
The current arduino_101_defconfig has the following:

 CONFIG_UART_QMSI_0_BAUDRATE=1000000
 CONFIG_UART_QMSI_0_HW_FC=y

Shouldn't arduino_101.dts default to those as well for uart0? Where else will the right values get set? UART0 is the UART between the Quark SE and the nRF51, so it's critical for BLE functionality that the configuration is correct.